app/src/main/java/com/kamron/pogoiv/Pokefly.java

```diff
@@ -451,6 +447,10 @@ private void unwatchScreen() {
         screenScanHandler.removeCallbacks(screenScanRunnable);
         screenScanRunnable = null;
         screenScanHandler = null;
+        if (screen != null) {
+            screen.exit();
+            screen = null;
+        }
```

Comment:
I'd prefer it isn't inside this function. Call from onDestroy directly or something? This is just about watch/unwatching the touch events.